lms/services/organization.py

```diff
@@ -146,8 +169,73 @@ def update_organization(self, organization, name=None, enabled=None, notes=None)
         if notes is not None:
             organization.settings.set("hypothesis", "notes", notes)
 
+        # Use the fact the default is ... to allow us to blank or set based on
+        # the param being present
+        if parent_public_id is None:
+            organization.parent_id = None
+        elif parent_public_id is not ...:
+            self._move_organization_parent(organization, parent_public_id)
+
         return organization
 
+    def _move_organization_parent(self, organization: Organization, parent_public_id):
+        """Change an organizations parent, without creating loops"""
+
+        # This would be caught by the next check, but doing it here means we
+        # can give a more sensible error message
+        if parent_public_id == organization.public_id:
+            raise InvalidOrganizationParent(
+                "Cannot set an organization to be it's own parent"
+            )
+
+        # Get a list including our self and all are children etc.
+        for item in self._get_hierarchy(organization.id):
+            if item.public_id == parent_public_id:
+                raise InvalidOrganizationParent(
+                    f"Cannot use '{parent_public_id}' as a parent as it a "
+                    "child of this organization"
+                )
+
+        parent = self._organization_search_query(
+            public_id=parent_public_id
+        ).one_or_none()
+        if not parent:
+            raise InvalidOrganizationParent(
+                f"Could not find parent organization: '{parent_public_id}'"
+            )
+
+        organization.parent = parent
+
+    def _get_hierarchy(self, id_) -> List[Organization]:
```

Comment:
We are not in the "save a query" mindset at all in for the admin pages but if we split this in two `_get_hierarchy_pk` and `_get_hierarchy` we could do a simpler `parent_public_id in _get_hierarchy_pk()` in `_move_organization_parent`

Reply:
Yes, this makes sense. We don't need to query here for the actual items. We need that in the next PR but we can add it on.